### PR TITLE
Add digicert at resources to allow migrate IoTHub to new iothub root …

### PR DESCRIPTION
**Pull request by PepLluis**
Add DigiCertcertificateGlobalRootG2 certificate

## Description

This sample need to be update for two reason, first one to meet requirements for new root cercificates for Azure IoTHub, the second one becouse de signature for mqtt librarty (5.0) has changed and need two extra parameters. "Content Type" and userProperties.

## Motivation and Context
Keeping running this sample free of build errors.
Allow migrate IoTHub to new root.

## How Has This Been Tested?

Yes. Both certificates have been tested on an ESP32
The BCM280 has not ben tested.

## Types of changes

- [ ] Improvement (non-breaking change that improves a sample)
- [x] Bug fix (fixes an issue with a current sample)
- [ ] New Sample (adds a new sample)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nanoframework/Samples?pullRequest=278) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nanoframework/Samples?pullRequest=278) before we can accept your contribution.<br/><br/>:x: PepLluis [sign now](https://cla.dotnetfoundation.org/nanoframework/Samples?pullRequest=278)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nanoframework/Samples?pullRequest=278) it.</sub>

File: samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Resources.Designer.cs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 //------------------------------------------------------------------------------
 // <auto-generated>
-//     This code was generated by a tool.
-//     Runtime Version:4.0.30319.42000
+//     Este código fue generado por una herramienta.
```
Comment (Ellerbach) [inline samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Resources.Designer.cs:3]: I know it's auto generated but can you leave the English comments?

File: samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs [hunk 2]
```diff
@@ -130,7 +135,7 @@
         );
 
         Trace("Getting twin properties");
-        mqttc.Publish($"{TwinDesiredPropertiesTopic}?$rid={Guid.NewGuid()}", Encoding.UTF8.GetBytes(""), MqttQoSLevel.AtLeastOnce, false);
+        mqttc.Publish($"{TwinDesiredPropertiesTopic}?$rid={Guid.NewGuid()}", Encoding.UTF8.GetBytes(""),string.Empty, new System.Collections.ArrayList(), MqttQoSLevel.AtLeastOnce, false);
```
Comment (Ellerbach) [inline samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs:138]: why do you need to change this?
Also please do a ctrl + k +d for the formatting, there is a little space missing

File: samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs [hunk 3]
```diff
@@ -145,7 +150,7 @@
         }
 
         Trace("Sending twin properties");
-        mqttc.Publish($"{TwinReportedPropertiesTopic}?$rid={Guid.NewGuid()}", Encoding.UTF8.GetBytes($"{{\"Firmware\":\"nanoFramework\",\"TimeToSleep\":{minutesToGoToSleep}}}"), MqttQoSLevel.AtLeastOnce, false);
+        mqttc.Publish($"{TwinReportedPropertiesTopic}?$rid={Guid.NewGuid()}", Encoding.UTF8.GetBytes($"{{\"Firmware\":\"nanoFramework\",\"TimeToSleep\":{minutesToGoToSleep}}}"),string.Empty,new System.Collections.ArrayList(), MqttQoSLevel.AtLeastOnce, false);
```
Comment (Ellerbach) [inline samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs:153]: same question here

File: samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs [hunk 4]
```diff
@@ -163,7 +168,7 @@
         Trace($"Pressure: {readResult.Pressure.Hectopascals}hPa");
 
         //Publish telemetry data
-        messageID = mqttc.Publish(telemetryTopic, Encoding.UTF8.GetBytes($"{{\"Temperature\":{readResult.Temperature.DegreesCelsius},\"Pressure\":{readResult.Pressure.Hectopascals}}}"), MqttQoSLevel.ExactlyOnce, false);
+        messageID = mqttc.Publish(telemetryTopic, Encoding.UTF8.GetBytes($"{{\"Temperature\":{readResult.Temperature.DegreesCelsius},\"Pressure\":{readResult.Pressure.Hectopascals}}}"),string.Empty, new System.Collections.ArrayList(), MqttQoSLevel.ExactlyOnce, false);
```
Comment (Ellerbach) [inline samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs:171]: and same question here :-)

**Review by Ellerbach — COMMENTED** [review]
Thanks a lot for the adjustments, few questions but all up, all good for the cert change and the comment in the code as well. It make things clear.

File: samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs (hunk 2 above)
Comment (PepLluis) [reply samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs:138]: As I explain in pull request, the signature for this function has changed... please check the mqtt library version.

**Comment by PepLluis** [comment]
Let me know if all is alright.
Thanks!

File: samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs (hunk 2 above)
Comment (Ellerbach) [reply samples/AzureMQTTTwinsBMP280Sleep/AzureMQTT/Program.cs]: you're right. I though I kept a compatible one...
You can simplify and pass null objects, no reason to create those. The code behind takes care of it: https://github.com/nanoframework/nanoFramework.m2mqtt/blob/27ee3891c54842c8a62565c26e8f67bbe7648bb9/M2Mqtt/MqttClient.cs#L2494

**Comment by Ellerbach** [comment]
@PepLluis you have some conflicts. You should resolve them and take care of having the right versions everywhere.

**Comment by PepLluis** [comment]
Well... some process was started in main between my fork and the PR... "[Update 11 NuGet dependencies](https://github.com/nanoframework/Samples/commit/955e3f5714c6f8c5079162d9b38b9cbe3063d6f9)"
This is breaking a Iot.Device.Bmxx80 dependency who is using UnitsNet.... with version 4.150
Let me know when this are solved to finish this PR.

image

**Pull request merged**